From: http://gmplib.org:8000/gmp-5.0/raw-rev/794410151f5f

Tweaked to remove the hunk in the Changelog, as it did not apply cleanly
( and who needs Changelogs? ;-) )

# HG changeset patch
# User Torbjorn Granlund <tege@gmplib.org>
# Date 1267122532 -3600
# Node ID 794410151f5f966bcb5c3489b6441614990efe7c
# Parent  948660e2e56d9cfaae035082b8fd473985505fb6
Fix a test case to work for long long limbs.

diff -r 948660e2e56d -r 794410151f5f tests/mpz/t-perfpow.c
--- a/tests/mpz/t-perfpow.c	Thu Feb 25 16:08:21 2010 +0100
+++ b/tests/mpz/t-perfpow.c	Thu Feb 25 19:28:52 2010 +0100
@@ -2,7 +2,7 @@
 
    Contributed to the GNU project by Torbjorn Granlund and Martin Boij.
 
-Copyright 2008, 2009 Free Software Foundation, Inc.
+Copyright 2008, 2009, 2010 Free Software Foundation, Inc.
 
 This file is part of the GNU MP Library.
 
@@ -109,7 +109,8 @@
 {
   mpz_t n, np, temp, primes[NRP];
   int i, j, k, unique, destroy, res;
-  unsigned long int nrprimes, primebits, g, exp[NRP], e;
+  unsigned long int nrprimes, primebits;
+  mp_limb_t g, exp[NRP], e;
   gmp_randstate_ptr rands;
 
   rands = RANDS;

